xquant/common/framework_report_utils.py

```diff
@@ -0,0 +1,259 @@
+#  Copyright 2024 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#  ==============================================================================
+
+import json
+import os
+from typing import Tuple, Any, Callable, Dict
+
+from model_compression_toolkit.core.common import Graph
+from model_compression_toolkit.core.common.visualization.tensorboard_writer import TensorboardWriter
+from xquant import XQuantConfig
+from xquant.common.constants import CS_METRIC_NAME, SQNR_METRIC_NAME, MSE_METRIC_NAME, REPORT_FILENAME
+from xquant.logger import Logger
+
+DEFAULT_METRICS_NAMES = [CS_METRIC_NAME,
+                         MSE_METRIC_NAME,
+                         SQNR_METRIC_NAME]
+
+class FrameworkReportUtils:
+
+    def __init__(self,
+                 tb_writer: TensorboardWriter):
+        """
+        Args:
+            tb_writer: TensorboardWriter object to add data to report (like graph and histograms).
+        """
+        Logger.info(f"Please run: tensorboard --logdir {tb_writer.dir_path}")
+        self.tb_writer = tb_writer
+
+    def add_histograms_to_tensorboard(self,
+                                      model: Any,
+                                      repr_dataset: Callable):
+        """
+        Collect histograms and add them to Tensorboard.
+
+        Args:
+            model: Model to collect histograms on.
+            repr_dataset: Dataset that is used an input for collecting histograms.
+
+        Returns:
+            None
+        """
+        Logger.critical(f"add_histograms_to_tensorboard is not implemented.")
+
+    def get_quantized_graph(self,
+                            quantized_model: Any,
+                            repr_dataset: Callable):
+        """
+        Get a graph representation of the quantized model.
+
+        Args:
+            quantized_model: The quantized model.
+            repr_dataset: Representative dataset to use during the graph building.
+
+        Returns:
+            Graph representation of the quantized model.
+        """
+
+        Logger.critical(f"get_quantized_graph is not implemented.")
+
+    def get_quant_graph_with_metrics(self,
```

Comment:
General comment for any reference of "metrics" throughout the code:

all metrics that you are currently referring to are "similarity" metrics right?
if so, I think it is important to add the word similarity wherever the "metrics" term is used (in the documentation and functions names).
When this project will grow and include more capabilities and more stuff that we would compute on the activation tensors, the term "metrics" would be ambiguous